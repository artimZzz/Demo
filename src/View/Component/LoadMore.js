import React, { Component, PureComponent } from 'react';
import { StyleSheet, Animated, View, Easing } from 'react-native';

export default class LoadMore extends PureComponent {

    /* constructor */
    constructor(props) {
        super(props);
        this.state = { animation: new Animated.Value(0) }
        this.loadingMore = false;
        this.animating = false;
    }

    /* render */
    render() {
        let opacity = this.state.animation.interpolate(
            {
                inputRange: [0, 0.5, 1],
                outputRange: [0, 1, 0]
            }
        );
        return (
            <Animated.View style={[styles.loadMore, { opacity: opacity }]}>
                {this.props.children}
            </Animated.View>
        );
    }

    /* public */
    startAnimation() {
        if (!this.animating) {
            this.animating = true;
            Animated.timing(this.state.animation).stop();
            this.performAnimation();
        }
        this.loadingMore = true;
    }
    
    stopAnimation() {
        this.loadingMore = false;
    }

    /* private */
	performAnimation() {
		this.state.animation.setValue(0);
		Animated.timing(
			this.state.animation,
			{ toValue: 1.0, duration: 1000, easing: Easing.linear }
		).start(
			() => {
				if (this.loadingMore) {
					this.performAnimation();
                }
                else {
                    this.animating = false;
                }
			}
		);
	}
}

/* style */
const styles = StyleSheet.create(
    {
        loadMore: {
            width: '100%', alignItems: 'center', paddingVertical: 10
        }
    }
);