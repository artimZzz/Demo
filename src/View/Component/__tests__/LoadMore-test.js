// __tests__/LoadMore-test.js
import React from 'react';
import LoadMore from '../LoadMore';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<LoadMore />).toJSON();
    expect(tree).toMatchSnapshot();
});