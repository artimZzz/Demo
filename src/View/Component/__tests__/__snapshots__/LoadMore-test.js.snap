// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "opacity": 0,
      "paddingVertical": 10,
      "width": "100%",
    }
  }
/>
`;
